docs/testing-with-codeception.md

```diff
@@ -0,0 +1,746 @@
+
+```php
+protected function activate( AcceptanceTester $I ) {
+    $this->loginAsAdmin();
+    $this->activatePlugin( 'some-plugin' );
+}
+
+protected function cleanup( AcceptanceTester $I ) {
+    $I->deactivatePlugin( 'some-plugin' );
+    $I->logout();
+}
+
+/**
+ * @before activate
+ * @before anotherPrerequisite
+ * @after cleanup
+ */
+public function checkPluginPageExists( AcceptanceTester $I ) {
+    // ...
+}
+```
+
+#### Environment
+
+Codeception tests can be instructed to run in multiple / different environments, via the `@env` special annotation, eg:
+
+```php
+/**
+ * @env chrome
+ * @env firefox
+ */
+public function someTest() {}
+```
+
+### Available modules
+
+Altis' Codeception integration comes bundled with `wp-browser` library, which provides additional modules to simplify testing WordPress applications. Altis pre-configures such modules via the zero-config installation, so you don't need to manually configure them unless you need to override some of the default values, which you can do via test suite configuration, eg:
+
+```yml
+# acceptance.suite.yml
+
+actor: AcceptanceTester
+modules:
+    enabled:
+        - WPDb
+        - WPBrowser
+        - \Helper\Acceptance
+    config:
+        WPBrowser:
+            headers:
+                X_WPBROWSER_REQUEST: 1
+```
+
+For a list of available modules, please check [`wp-browser` documentation on modules](https://wpbrowser.wptestkit.dev/modules), and the respective configuration options, and methods, of each.
+
+These are the available modules from wp-browser:
+
+#### WPBrowser
+
+This module extends the `PHPBrowser` module, adding WordPress-specific configuration parameters and methods. It simulates a user interaction with the site without Javascript support; if you need to test your project with Javascript support use the `WPWebDriver` module instead.
+
+[Read more on WPBrowser module configuration](https://wpbrowser.wptestkit.dev/modules/wpbrowser#configuration).
+
+#### WPWebDriver
+
+This module extends the `WebDriver` module adding WordPress-specific configuration parameters and methods. It simulates a user interaction with the site with Javascript support; if you don't need to test your project with Javascript support use the WPBrowser module to skip the overhead of loading a headless browser.
+
+Altis comes with built-in browser support for Chrome, and Firefox, based on Selenium standalone Docker images, which is pre-configured to run with and be available for acceptance tests with zero-configuration required.
+
+**Important notes:**
+
+- During acceptance tests, two processes (or more) are working in parallel:
+a. The test runner request, ie: Codeception process.
+b. The browser session driven by `WPWebDriver`, ie: application process.
+Both of those use different configurations and different running context / environment. And it'll save you time to distinguish between the two running processes/threads.
```

Comment:
Should the a. and bb. be indented here?

Reply:
Yeah, but maybe a nested list is more consistent, will update.

Reply:
```suggestion
- During acceptance tests, two processes (or more) are working in parallel:
  - The test runner request, ie: Codeception process.
  - The browser session driven by `WPWebDriver`, ie: application process.
Both of those use different configurations and different running context / environment. And it'll save you time to distinguish between the two running processes/threads.
```